qp_klp/tests/test_step.py

```diff
@@ -602,3 +680,155 @@ def test_get_tube_ids_from_qiita(self):
+
+    def test_generate_commands(self):
+        self._create_test_input(4)
+
+        fake_client = FakeClient()
+
+        # self.pipeline represents a metagenomic pathway.
+        step = Step(self.pipeline, self.qiita_id, None)
+
+        # need to generate some metadata in order to generate commands.
+        step.generate_special_map(fake_client)
+
+        # test base _generate_commands() method; contains only commands used
+        # across all pipeline types.
+        step.generate_commands()
+
+        exp = [
+            (f'cd {self.output_file_path}; '
+             'tar zcvf logs-ConvertJob.tgz ConvertJob/logs'),
+            (f'cd {self.output_file_path}; '
+             'tar zcvf reports-ConvertJob.tgz ConvertJob/Reports '
+             'ConvertJob/logs'),
+            (f'cd {self.output_file_path}; '
+             'tar zcvf logs-FastQCJob.tgz FastQCJob/logs'),
+            (f'cd {self.output_file_path}; '
+             'tar zcvf logs-QCJob.tgz QCJob/logs'),
+            (f'cd {self.output_file_path}; '
+             'tar zcvf reports-FastQCJob.tgz FastQCJob/fastqc'),
+            (f'cd {self.output_file_path}; '
+             'tar zcvf logs-GenPrepFileJob.tgz GenPrepFileJob/logs'),
+            (f'cd {self.output_file_path}; '
+             'tar zcvf prep-files.tgz GenPrepFileJob/PrepFiles'),
+            (f'cd {self.output_file_path}; '
+             'mv failed_samples.html final_results'),
+            (f'cd {self.output_file_path}; '
+             'tar zcvf reports-QCJob.tgz QCJob/Feist_11661/fastp_reports_dir '
+             'QCJob/Gerwick_6123/fastp_reports_dir '
+             'QCJob/NYU_BMS_Melanoma_13059/fastp_reports_dir'),
+            (f'cd {self.output_file_path}; '
+             'tar zcvf sample-files.tgz 211021_A00000_0000_SAMPLE_NYU_BMS_'
+             'Melanoma_13059_blanks.tsv 211021_A00000_0000_SAMPLE_Feist_'
+             '11661_blanks.tsv 211021_A00000_0000_SAMPLE_Gerwick_6123_'
+             'blanks.tsv'),
+            (f'cd {self.output_file_path}; '
+             'mv 211021_A00000_0000_SAMPLE_NYU_BMS_Melanoma_13059_blanks.tsv '
+             'BASE_DIRECTORY/qp_klp/tests/data'
+             '/QDir/uploads/13059'),
+            (f'cd {self.output_file_path}; '
+             'mv 211021_A00000_0000_SAMPLE_Feist_11661_blanks.tsv '
+             'BASE_DIRECTORY/qp_klp/tests/data/QDir/'
+             'uploads/11661'),
+            (f'cd {self.output_file_path}; '
+             'mv 211021_A00000_0000_SAMPLE_Gerwick_6123_blanks.tsv '
+             'BASE_DIRECTORY/qp_klp/tests/data/QDir/'
+             'uploads/6123'),
+            (f'cd {self.output_file_path}; (find *.tgz -maxdepth 1 -type f '
+             '| xargs mv -t final_results) || true')]
+
+        # replace unique string w/the base-directory path in the expected
+        # output.
+        for i in range(0, len(exp)):
+            exp[i] = exp[i].replace('BASE_DIRECTORY', getcwd())
+
+        self.assertEqual(set(step.cmds), set(exp))
```

Comment:
Why do they need to be converted to a set? It should test that they are the same, no?

Reply:
Good catch. Previously the order of some individual operations was undetermined from macosx to linux because of (I believe) walk(). As it's now fixed in order, we can test the order as well.